### DISCUSSION TITLE
Uso esperado al file system

### DISCUSSION THREAD

**Discussion by julchat** [dds-jv/foro#131] [Consultas de objetos] [answered]
Hola buenas noches, una duda
¿Cuál es el uso que le van a dar el filesystem? Lo vamos a usar simplemente para meter todos los caracteres de un archivo en un buffer o del buffer al archivo de una o el usuario debería también poder hacer varias lecturas/escrituras en serie partiendo desde donde dejamos el flujo la última vez?
Y al usuario realmente le va a interesar en que byte del buffer guardar la información o basta con simplificar las cosas y esperar unicamente cuántos bytes el usuario quiere leer o escribir?
Tengo sueño, espero la respuesta, gracias, hasta la proximaaa

**Comment by gastonprieto** [comment, marked as answer]
Buenas!

No entendí muy bien esto:

> Lo vamos a usar simplemente para meter todos los caracteres de un archivo en un buffer o del buffer al archivo de una?

Pero las operaciones son:
> escribir sincrónicamente un bloque de n bytes de archivo
leer sincrónicamente un bloque de n bytes de un archivo
leer asincrónicamente un bloque de n bytes de un archivo

> usuario debería también poder hacer varias lecturas/escrituras en serie partiendo desde donde dejamos el flujo la última vez

En principio no esta dentro de lo obligatorio, pero si parece ser mas importante en los requerimientos opcionales.

> Y al usuario realmente le va a interesar en que byte del buffer guardar la información o basta con simplificar las cosas y esperar unicamente cuántos bytes el usuario quiere leer o escribir?

Parece ser caso de uso valido para cuando escribis secuencialmente, pero que pasa si queres escribir en una parte particular del archivo?

----------
Una cosa importante es que vos estas mejorando la interfaz, y le estarías agregando funcionalidades, pero no debería perder las operaciones que podía hacer con la interfaz original (o sea, tiene que soportar los mismos Casos de uso)

**Comment by gastonprieto** [comment]
@julchat te sirvio? te quedo alguna duda?

**Comment by julchat** [comment]
Sisi estoy trabajando en el ejercicio, cualquier cosa pregunto
